packages/@sanity/client/src/config.js

```diff
@@ -89,13 +76,27 @@ exports.initConfig = (config, prevConfig) => {
     const cdnHost = newConfig.isDefaultApi ? defaultCdnHost : host
 
     if (newConfig.useProjectHostname) {
-      newConfig.url = `${protocol}://${newConfig.projectId}.${host}/v1`
-      newConfig.cdnUrl = `${protocol}://${newConfig.projectId}.${cdnHost}/v1`
+      newConfig.url = `${protocol}://${newConfig.projectId}.${host}/v${newConfig.apiVersion}`
+      newConfig.cdnUrl = `${protocol}://${newConfig.projectId}.${cdnHost}/v${newConfig.apiVersion}`
     } else {
-      newConfig.url = `${newConfig.apiHost}/v1`
+      newConfig.url = `${newConfig.apiHost}/v${newConfig.apiVersion}`
       newConfig.cdnUrl = newConfig.url
     }
   }
 
   return newConfig
 }
+
+exports.validateApiVersion = function validateApiVersion(apiVersion, nowOffset = 0) {
```

Comment:
We don't strictly need to validate this, since the API gateway will also validate it and generate an appropriate error message - but obviously feel free to include it if you feel like it's beneficial.